File: ISBN-10 Validation.py
import re
def valid_ISBN10(isbn): 
    if not(re.match(r'^\d{9}[0-9X]$',isbn)):
        return False
    return sum([int(isbn[i-1])*i if isbn[i-1]!="X" else 10*i  for i in range(1,11)])%11==0

File: ISBN-10 Validation.ts
export function validISBN10(isbn: string): boolean {
    if(!/^[0-9]{9}[0-9X]$/.test(isbn)){
      return false
    }
    return [...Array(10).keys()].map(i => i+1).map(v => isbn[v-1]!=='X'? v*Number(isbn[v-1]) : v*10).reduce((a,b) => a+b,0)%11==0
}
